common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -162,12 +174,24 @@ class HTTPClient(
             throw IOException(NetworkStrings.HTTP_RESPONSE_PAYLOAD_NULL)
         }
 
+        val verificationStatus = if (shouldSignResponse && nonce != null) {
+            verifyResponse(path, connection, payload, nonce)
+        } else {
+            HTTPResult.VerificationStatus.NOT_VERIFIED
+        }
+
+        if (verificationStatus == HTTPResult.VerificationStatus.ERROR &&
```

Comment:
This should be
```suggestion
        if (verificationStatus != HTTPResult.VerificationStatus.VERIFIED &&
```

If for some reason it was not verified, it should also fail.
I'd add a test for that case too if possible since that's important.

Reply:
Oh good point, never mind then. iOS does the same for that reason 👍🏻